src/abstractindsnetwork.jl

```diff
@@ -4,6 +4,45 @@ abstract type AbstractIndsNetwork{I} <:
 # Field access
 data_graph(graph::AbstractIndsNetwork) = _not_implemented()
 
+# Overload if needed
+is_directed(::Type{<:AbstractIndsNetwork}) = false
+
 # AbstractDataGraphs overloads
 vertex_data(graph::AbstractIndsNetwork, args...) = vertex_data(data_graph(graph), args...)
 edge_data(graph::AbstractIndsNetwork, args...) = edge_data(data_graph(graph), args...)
+
+# 
+# Index access
+# 
+
+function uniqueinds(is::AbstractIndsNetwork, edge::AbstractEdge)
+  inds = IndexSet(get_assigned(is, src(edge), Index[]))
+  for ei in setdiff(incident_edges(is, src(edge)...), [edge])
+    inds = unioninds(inds, get_assigned(is, ei, Index[]))
+  end
+  return inds
+end
+
+function uniqueinds(is::AbstractIndsNetwork, edge::Pair)
+  return uniqueinds(is, edgetype(is)(edge))
+end
+
+# 
+# Convenience functions
+# 
+
+function Base.merge(is1::AbstractIndsNetwork{I}, is2::AbstractIndsNetwork{I}) where {I}
```

Comment:
The closest Graphs.jl function for this seems to be [`Graphs.union`](https://juliagraphs.org/Graphs.jl/dev/core_functions/operators/#Base.union-Union{Tuple{T},%20Tuple{T,%20T}}%20where%20T%3C:Graphs.SimpleGraphs.AbstractSimpleGraph).

So this could be implemented as a more generic function, where we overload `Graphs.union` on a `DataGraph` but then allow users to pass a custom function for how to merge the metadata. Then this functionality could be implemented through `Graphs.union` by passing `unioninds` as the custom metadata merging function. We don't need to implement that in this PR, I'm just trying to think about what the more general principle or functionality behind this function might be since that might inspire a principled name to use here (i.e. the design in terms of `Graphs.union` might lead to calling this function as `union(unioninds, is1, is2)`, for example).

Reply:
Sounds good. For now I can just rename it to specifically overload `Graphs.union` for `IndsNetwork`s here, which can then be replaced by a more general interface for once it is implemented for `DataGraph`s? Or would you already like to add the custom metadata merging function?

Reply:
Maybe we should name it something more specific for now like `union_all_inds` and think about the more general interface in a future PR.